zoo/analytics/tasks/docker.py

```diff
@@ -1,41 +1,44 @@
+from pathlib import Path
+
 from dockerfile_parse import DockerfileParser
 
-from . import Language, OS
+from . import DockerImage, OS
 from ..models import DependencyType
-from .utils import parse_version
+from .utils import DockerImageId
 
 KNOWN_DEPS = {
     DependencyType.LANG: {"python", "node"},
     DependencyType.OS: {"alpine", "debian"},
 }
 KNOWN_DEBIAN_IMAGES = {"python", "node"}
 
+# TODO: make more clever ways of dealing with Dockerfile paths.
+DOCKERFILE_PATHS = ["/", "/docker/"]
+
 
 def analyze(repository, path):
     """Parse dockerfile for image information about platform."""
-    dockerfile = path / "Dockerfile"
-    if not dockerfile.is_file():
-        return
-    dockerfile.resolve()
-    dfp = DockerfileParser(path=dockerfile.as_posix())
-    for command in dfp.structure:
-        if command["instruction"] != "FROM":
+    for sub_path in DOCKERFILE_PATHS:
+        dockerfile = Path(str(path) + sub_path + "/Dockerfile")
+        if not dockerfile.is_file():
             continue
-        yield from parse_base_image_name(command["value"], repository=repository)
+        dockerfile.resolve()
+        dfp = DockerfileParser(path=dockerfile.as_posix())
 
+        for command in dfp.structure:
+            if command["instruction"] != "FROM":
+                continue
+            parsed_image_id = DockerImageId(command["value"])
 
-def parse_base_image_name(base_image: str, repository):
-    lang_fragment, _, os_fragment = base_image.partition("-")
-    for lang_name in KNOWN_DEPS[DependencyType.LANG]:
-        if lang_name not in lang_fragment:
-            continue
-        lang_version = parse_version(lang_fragment.replace(lang_name, "").strip(": "))
-        yield Language(lang_name, lang_version, for_production=True)
-
-        if os_fragment == "alpine":
-            yield OS("alpine", for_production=True)
-        elif os_fragment.startswith("alpine"):
-            os_version = parse_version(os_fragment.replace("alpine", "").strip(": "))
-            yield OS("alpine", os_version, for_production=True)
-        elif lang_name in KNOWN_DEBIAN_IMAGES:
-            yield OS("debian", for_production=True)
+            yield DockerImage(
+                name=parsed_image_id.repository_name,
+                version=parsed_image_id.version,
+                for_production=True,
```

Comment:
What does `for_production` do ? And why is it `True` without determining it somehow ? 

Im not saying we should change it, just curious.

Reply:
I looked into the codebase and it seems that this flag is used for determining whether a dependency is used in production or not, i.e. when it parses `requirements.txt` for python libs it separates "for production" ones as coming from `requirements.txt` and "not for production" ones coming from other requirement files, e.g. `test-requirements.txt`. So in this case we dont have any clever parsing whether a dockerfile is or is not "for production" so it simply sets it to True every time, as that is typically the case.

Correct me if I am wrong in my assumptions please @aexvir .

Reply:
Yep, that's correct. We could do some smart categorization based on if it's `Dockerfile` (for prod) and `Dockerfile.whatever` (non prod) but I don't think it's on the scope of the issue you are trying to solve here, and it would raise false positives in projects like our `booking/backend`. So for now this looks good to me 🙂

Reply:
yes agree, out of scope now. thanks guys

btw i find it a bit misleading as having `for_production=True` for package, because its in `requirements.txt` is weird as the package/service do not need to be in production. i'd maybe just store the name of the file it was parsed from and think about categorazing it later (e.g. - core, test, docs, dev)